src/components/searchBar/Results/index.tsx

```diff
@@ -0,0 +1,173 @@
+          onClick={handleFilter}
+        >
+          Blocks ({blockNumber ? 1 : 0})
+        </Button>
+        <Button
+          isActive={filter === 'extrinsics'}
+          type="extrinsics"
+          onClick={handleFilter}
+        >
+          Extrinsics ({extrinsics?.length || 0})
+        </Button>
+      </div>
+      <PDScrollArea
+        verticalScrollClassNames="py-4"
+        verticalScrollThumbClassNames="before:bg-dev-purple-700 dark:before:bg-dev-purple-300"
+        className="flex max-h-80 flex-col gap-4"
+      >
+        {showBlock && blockNumber && (
+          <>
+            <p className={cn(
+              'my-4 border-b p-1',
+              'font-geist text-dev-white-1000 font-body2-regular dark:text-dev-black-1000',
+            )}
+            >
+                Blocks ({blockNumber ? 1 : 0})
+            </p>
+            <PDLink
+              to={`/explorer/${blockNumber}`}
+              className={cn(
+                'flex w-full items-center justify-between',
+                'px-4 py-3.5',
+                'transition-[background] duration-300',
+                'hover:bg-dev-black-900 hover:dark:bg-dev-purple-200',
+              )}
+            >
+              <p className="font-geist text-dev-white-200 font-body2-regular dark:text-dev-black-1000">
+                  Block#
+                <span className="font-body2-bold"> {blockNumber}</span>
+              </p>
+              <p className="font-geist text-dev-white-1000 font-body2-regular dark:text-dev-black-1000">
+                  Block
+              </p>
+            </PDLink>
+          </>
+        )}
+
+        {filteredExtrinsics && (
+          <>
+            <p className={cn(
+              'my-4 border-b p-1',
+              'font-geist text-dev-white-1000 font-body2-regular dark:text-dev-black-1000',
+            )}
+            >
+                Extrinsics ({filteredExtrinsics?.length})
+            </p>
+
+            {filteredExtrinsics?.map((extrinsic) => (
+              <div
+                key={extrinsic?.id}
+                data-extrinsic-id={extrinsic?.id}
+                onClick={handleOpenModal}
+                className={cn(
+                  'flex w-full items-center justify-between',
+                  'cursor-pointer px-4 py-3.5',
+                  'transition-[background] duration-300',
+                  'hover:bg-dev-black-900 hover:dark:bg-dev-purple-200',
+                )}
+              >
+                <p className="font-geist text-dev-white-200 font-body2-regular dark:text-dev-black-1000">
+                  Extrinsic#
+                  <span className="font-body2-bold"> {extrinsic?.id}</span>
+                </p>
+                <p className="font-geist text-dev-white-1000 font-body2-regular dark:text-dev-black-1000">
+                  Extrinsic
+                </p>
+              </div>
+            ))}
+          </>
+        )}
+
+        {blockNumber || filteredExtrinsics ? null : (
```

Comment:
The condition looks a bit controversial to me

Reply:
Done